### PR TITLE
Allow plugin to filter and ignore conditions

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/883

## Type of change

*Please select one of the options below.*

- New feature (non-breaking change which adds functionality)

# Discussion

This feature is pretty useful to find the process group that blocks an operation e.g. during upgrades.

# Testing

local + unit

# Documentation

Added in the cli command.

# Follow-up

-

### CODE REVIEW

File: kubectl-fdb/cmd/analyze.go [hunk 1]
```diff
@@ -70,6 +70,11 @@ func newAnalyzeCmd(streams genericclioptions.IOStreams) *cobra.Command {
 				return err
 			}
 
+			ignoreConditions, err := cmd.Flags().GetStringArray("ignore-condition")
```
Comment (JoshuaMcManus) [inline kubectl-fdb/cmd/analyze.go:73]: Should we add a validation to the arguments such that it's within ProcessGroupConditionType.AllProcessGroupConditionTypes()?

**Review by JoshuaMcManus — COMMENTED** [review]
One other suggestion that _might_ be useful is if we were to report how many conditions were ignored as one line in the output.

File: kubectl-fdb/cmd/analyze.go (hunk 1 above)
Comment (johscheuer) [reply kubectl-fdb/cmd/analyze.go:73]: I think that's a good idea and we should fail early if a condition is passed that is unknown.

**Review by brownleej — APPROVED** [review]
👍

**Review by JoshuaMcManus — APPROVED** [review]

**Pull request merged**